src/Objects/Update.php

```diff
@@ -118,4 +120,37 @@ public function hasCommand(): bool
             ->flatten()
             ->contains('type', 'bot_command');
     }
+
+    public function getAllEntities(): ?array
+    {
+        return $this->getMessage()->{$this->getEntitiesKey()};
+    }
+
+    public function getCommandEntities(): ?Collection
+    {
+        return collect($this->getAllEntities())->filter(fn (MessageEntity $entity) => $entity->type === 'bot_command');
+    }
+
```

Comment:
Is this the update object's responsibility? I think this can be safely moved to the parser or the command bus.

Reply:
Initially this code was in the command bus.

Then I found myself repeating code in the Parser...or passing the Update object back and forth to get the data I needed.

I (believe) I found it much cleaner letting the update object hold/provide this information. But keen to see how to refactor this if you think otherwise.

A good example is filtering for command entities.

We need that in the parser AND in the command bus

https://github.com/telegram-bot-sdk/telegram-bot-sdk/blob/master/src/Commands/CommandBus.php#L148
And
https://github.com/telegram-bot-sdk/telegram-bot-sdk/blob/master/src/Commands/Parser.php#L218-L221

I found that by just asking the update object, "could I have your commandEntities" that it felt better. But I'm still only learning!

Reply:
Where did my last comment go?!

Reply:
Ok

I've looked again and keep having the same issues. Both parser and command bus need to do the same stuff and so I'm duplicating code.

If I keep it on the update it DRYs it up very well.

Please show me when you can what I should be doing!